src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -186,14 +188,29 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
 
                         module.ModuleName = new string(chars, 0, length);
 
-                        length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                        for (; ; )
+                        {
+                            length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                            if (length == chars.Length)
+                            {
+                                char[] toReturn = chars;
+                                chars = null;
+                                ArrayPool<char>.Shared.Return(toReturn);
+                                minimumLength = Math.Min(minimumLength * 2, short.MaxValue);
```

Comment:
Should the new length be rather based on `chars.Length`? `chars.Length` may be a lot more than `minimumLength`. The code may just keep iterating unnecessarily several times to catch up.